### ISSUE TITLE
Burrow 3D fill on hidden tiles does not fill or erase across z-levels

### ISSUE THREAD

**Issue by myk002** [DFHack/dfhack#3979]
When the flood area is the hidden walls

**Comment by xBlackTalonX** [comment]
tl;dr - burrow 3D fill doesn't **fill** across z-levels of hidden walls either

After a good discussion [here](https://discord.com/channels/793331351645323264/1017471248567124049/1171848712994168884), I think burrow flood fill has a primary function on walkable tiles and a secondary function on hidden tiles and open space. For this secondary function, fill applied on hidden tiles and open space, there are some inconsistencies to address

- Burrow fill/erase on **open space**, all connected **open space** tiles are filled
  - 2D on that z-level - Test PASS
  - 3D across z-levels - Test PASS
- Burrow fill/erase on **hidden**, all connected **hidden** tiles are filled
  - 2D on that z-level - Test PASS
  - 3D across z-levels - Test FAIL (behavior identical to 2D)

Suggest changing title to, Burrow 3D fill on hidden tiles does not fill or erase across z-levels. Or open new issue and close this as duplicate or whatever your issue management preferences are. :-D
